certbot-ci/certbot_integration_tests/nginx_tests/context.py

```diff
@@ -1,5 +1,83 @@
+import os
+import subprocess
+import random
+import contextlib
+import ssl
+
 from certbot_integration_tests.certbot_tests import context as certbot_context
+from certbot_integration_tests.utils import misc
+from certbot_integration_tests.nginx_tests import nginx_config as config
 
 
 class IntegrationTestsContext(certbot_context.IntegrationTestsContext):
     """General fixture describing a certbot-nginx integration tests context"""
+    def __init__(self, request):
+        super(IntegrationTestsContext, self).__init__(request)
+
+        self.nginx_root = os.path.join(self.workspace, 'nginx')
+        os.mkdir(self.nginx_root)
+
+        self.webroot = os.path.join(self.nginx_root, 'webroot')
+        os.mkdir(self.webroot)
+        with open(os.path.join(self.webroot, 'index.html'), 'w') as file_handler:
+            file_handler.write('Hello World!')
+
+        self.other_port = random.randint(6000, 6999)
+
+        self.key_path, self.cert_path = config.create_self_signed_certificate(self.nginx_root)
+
+        self.nginx_config_path = os.path.join(self.nginx_root, 'nginx.conf')
+        self.nginx_config = None
+
+    @contextlib.contextmanager
```

Comment:
Why do you not like having a context manager on this method ?

Because I disagree here. The context manager is helping a lot to simplify the complexity, and improve the readability of the test. Other solutions will either complexify the context object, with weird and useless reloading of nginx, or the test module itself with more assertions or more fixtures.

Reply:
As an alternative, since the integration context is used only once thanks to the test parametrization, I propose to merge all its logic into the test method itself, and remove this fixture.

Reply:
> Why do you not like having a context manager on this method ?

I just feel like it defeats some of the benefits of the fixture. Instead of letting it automatically handle setting up and tearing down the context needed for the tests, if you ignore the behavior we're getting from the Certbot context object, we're doing most of the work when we use this context manager. Why have both the new fixture and the context manager?

I'm OK with your alternative, as long as all of the logic isn't put directly inline in the parameterized test method. I don't want to make a massive function or give ourselves significant work if we want to add a test in the future that doesn't exactly fit with the current method we have.

I don't necessarily prefer this over removing the new fixture all together, but yet another thing we could do is https://docs.pytest.org/en/latest/example/parametrize.html#apply-indirect-on-particular-arguments.

Reply:
The indirect functionality is precisely what I was searching, the reason of the workaround I used with this context manager to keep things elegant.

Implemented. I use the indirection to configure the context fixture. The context manager is removed, and nginx is setup with the fixture creation.